source/javascripts/components/common/RepoYmlStorageActions.tsx

```diff
@@ -1,18 +1,59 @@
-import { useState, useMemo } from "react";
+import { useState, useEffect, useRef } from "react";
 import { Box, Icon, Text, Notification, Link } from "@bitrise/bitkit";
 import CopyToClipboard from "react-copy-to-clipboard";
 import { AppConfig } from "../../models/AppConfig";
 import appConfigAsYml from "../../utils/appConfigAsYml";
+import useMonolithApiCallback from "../../hooks/api/useMonolithApiCallback";
 
 type RepoYmlStorageActionsProps = {
 	appConfig: AppConfig | string;
 };
 
+const useFormattedYml = (appConfig: AppConfig): { yml: string; failed: boolean; loading: boolean } => {
+	const [yml, setYml] = useState(typeof appConfig === "string" ? appConfig : "");
+	const formatAppConfigRef = useRef<() => void>();
+	const { loading, failed, result, call } = useMonolithApiCallback<string>(
+		"/api/cli/format",
+		{
+			method: "POST",
+			body: JSON.stringify(appConfig)
+		},
+		(value: string) => value
```

Comment:
I'm wondering if instead of this, we could pass a parameter to `useMonolithApiCallback`, something like an enum of `text` and `json`, which would then internally call either `text` or `json` on the return value of `fetch`? Feels a bit awkward to pass an identity function here just to override the default value.

Reply:
I'd prefer the flexibility of the parser function over something like just text/json. This way you can do additional processing on it. It's kind of analogous to a select function in react-query, at least that is my mental model about it.

Reply:
And on a side note, I would avoid enums in general.